src/compiler/abepi.rs

```diff
@@ -218,13 +218,29 @@ impl<F: From<u64> + Into<u32> + Clone, V: Clone> CompilationUnit<F, V> {
         }
     }
 
+    // my question here is do we really need this function?
+    // Can't we just negate the result of compile_expression_eq?
     fn compile_expression_neq(
         &self,
         _dsym: DebugSymRef,
         _lhs: Expression<F, V>,
         _rhs: Expression<F, V>,
     ) -> CompilationResult<F, V> {
-        todo!()
+        assert!(_lhs.is_arith());
+        assert!(_rhs.is_arith());
+
+        let lhs = self.compile_expression_airth(_lhs);
+        let rhs = self.compile_expression_airth(_rhs);
+
+        // In One Zero: 0 is false
+        // lhs != rhs => lhs - rhs == 0 is false
```

Comment:
If you substract two different numbers the result is 0?

Reply:
The comment was not so clear, fixed it